src/mono/mono/metadata/icall.c

```diff
@@ -2807,6 +2810,21 @@ ves_icall_RuntimeType_GetPacking (MonoQCallTypeHandle type_handle, guint32 *pack
 	}
 }
 
+gint8
+ves_icall_RuntimeType_GetCallingConventionFromFunctionPointerInternal (MonoQCallTypeHandle type_handle)
+{
+	MonoType *type = type_handle.type;
+	g_assert (type->type == MONO_TYPE_FNPTR);
+	return type->data.method->suppress_gc_transition ? MONO_CALL_UNMANAGED_MD : type->data.method->call_convention;
```

Comment:
Could you please explain how it would work after the clean up?

I had to do it this way because `ModifiedFunctionPointerType` expects to receive `0x09` in order to look for calling conventions in modifiers: https://github.com/dotnet/runtime/blob/9a5fa7445c47e65e1cd454a5255b671ed81b3ae4/src/libraries/System.Private.CoreLib/src/System/Reflection/ModifiedFunctionPointerType.cs#L56-L80